vscode/src/client.ts

```diff
@@ -273,6 +301,13 @@ export default class Client extends LanguageClient implements ClientInterface {
     this.#context = context;
     this.ruby = ruby;
     this.#formatter = "";
+
+    this.onNotification("delegate/textDocument/virtualState", (params) => {
```

Comment:
When we receive the custom notification, we remember the virtual source for the host language in a hash.